### PR TITLE
Restore test refactor

### PR DESCRIPTION
**Pull request by jmathias3**
Main issue that this update solves is that the kb is not able to add and delete follow up prompts that point to the same question in the same update. Deleting and then Adding is the process we need to use to change existing follow up prompts. In order to make changes to existing follow up prompts, we have to identify when they are occurring, process the delete step in one update and then process the addition of the new version in a second update. 

This update also includes extra formatting and updates to the Kb api readme

### CODE REVIEW

File: kb-node-client/index.js [hunk 1]
```diff
@@ -274,40 +274,51 @@ let QnAMakerAPI = function(config) {
 
             return operationDetails
 
-        }
-    }
-
-    let marshaling = {
```
Comment (KyleMit) [inline kb-node-client/index.js:280]: Tiny comment, but I most commonly see this with two ll's although - both seem acceptable

* [“Marshalling” or “Marshaling”?](https://stackoverflow.com/a/4434616/1366033)
* [Marshalling (computer science)](https://en.wikipedia.org/wiki/Marshalling_(computer_science))

File: faq-site/tools/utilities.js [hunk 2]
```diff
@@ -89,6 +89,15 @@ function slugify(title) {
...
 
+/**
+ * Test whether item is is empty object `{}`
+ * @param {object} obj 
+ * @description See also: https://stackoverflow.com/q/679915/1366033
+ */
+function isEmptyObj(obj) {
+    return JSON.stringify(obj) === JSON.stringify({});
+}
+
```
Comment (KyleMit) [inline faq-site/tools/utilities.js:100]: Local functions aren't exported by default, so you'll have to add it to the `module.exports` object at the top

**Comment by KyleMit** [comment]
Did you intend to delete `pollForOperationComplete` from the [latest commit](https://github.com/VermontDepartmentOfHealth/covid-bot/pull/47/commits/5308e32aa8a86ff204248f829fc2429616c811b9#diff-109a153241d35287a53af8a5d938d778L104)?  It looks like it's still in use

File: faq-site/tools/utilities.js (hunk 2 above)
Comment (KyleMit) [reply faq-site/tools/utilities.js:100]: hard to catch since we're not actually running this between commits during the day

**Comment by jmathias3** [comment]
I can't reply to your `pollForOperationComplete` comment for some reason, so adding a new one. I did intend to delete it. I moved it to the Operations object and never deleted the original implementation. I think all of the references point to the version on operations.

**Pull request merged**